ludwig/utils/backward_compatibility.py

```diff
@@ -598,15 +598,27 @@ def upgrade_missing_value_strategy(config: Dict[str, Any]) -> Dict[str, Any]:
     return config
 
 
+@register_config_transformation("0.6", ["trainer"])
+def _upgrade_max_batch_size(trainer: Dict[str, Any]) -> Dict[str, Any]:
+    if "increase_batch_size_on_plateau_max" in trainer:
+        trainer["max_batch_size"] = trainer.pop("increase_batch_size_on_plateau_max")
```

Comment:
We should check that `trainer["max_batch_size"]` is not already set. If so, we should discard `increase_batch_size_on_plateau_max` and log a warning.